SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialStrategySystem.cpp

```diff
@@ -1,58 +1,306 @@
+			}
+			AuthorityIntentView.Remove(Delta.EntityId);
+			PendingMigrations.Remove(Delta.EntityId);
+		}
+		break;
 		case EntityDelta::TEMPORARILY_REMOVED:
 		{
-			// TODO
+			for (auto& Storage : DataStorages)
+			{
+				Storage->OnRemoved(Delta.EntityId);
+			}
+			const SpatialGDK::EntityViewElement& Element = LBView.GetView().FindChecked(Delta.EntityId);
+			for (auto& Storage : DataStorages)
+			{
+				Storage->OnAdded(Delta.EntityId, Element);
+			}
 		}
 		break;
 		default:
 			break;
 		}
 	}
+
+	Strategy->Advance(Connection);
 }
 
-void SpatialStrategySystem::Flush(SpatialOSWorkerInterface* Connection)
+void FSpatialStrategySystem::Flush(ISpatialOSWorker& Connection)
 {
-	// TODO
+	PartitionsMgr->Flush(Connection);
+
+	if (bStrategySystemInterestDirty)
+	{
+		UpdateStrategySystemInterest(Connection);
+	}
+
+	Strategy->TickPartitions(*PartitionsMgr);
+	Strategy->Flush(Connection);
+
+	TSet<Worker_EntityId_Key> ModifiedEntities;
+	for (auto& Storage : DataStorages)
+	{
+		ModifiedEntities = ModifiedEntities.Union(Storage->GetModifiedEntities());
+	}
+
+	FMigrationContext Ctx(MigratingEntities, ModifiedEntities);
+
+	Strategy->CollectEntitiesToMigrate(Ctx);
+
+	for (auto& Storage : DataStorages)
+	{
+		Storage->ClearModified();
+	}
+
+	for (auto PendingMigration : PendingMigrations)
+	{
+		if (!Ctx.EntitiesToMigrate.Contains(PendingMigration.Key))
+		{
+			Ctx.EntitiesToMigrate.Add(PendingMigration);
+		}
+	}
+	PendingMigrations.Empty();
+
+	for (auto const& Migration : Ctx.EntitiesToMigrate)
+	{
+		Worker_EntityId EntityId = Migration.Key;
+
+		if (!ensureAlways(!MigratingEntities.Contains(EntityId)))
+		{
+			continue;
+		}
+
+		FPartitionHandle Partition = Migration.Value;
+		TOptional<Worker_PartitionId> DestPartition = PartitionsMgr->GetPartitionId(Partition);
+		AuthorityIntentV2& AuthIntent = AuthorityIntentView.FindChecked(EntityId);
+		if (!DestPartition)
+		{
+			PendingMigrations.Add(EntityId, Partition);
+			continue;
+		}
+		PendingMigrations.Remove(EntityId);
+		AuthIntent.PartitionId = DestPartition.GetValue();
+		AuthIntent.AssignmentCounter++;
+
+		ComponentUpdate Update(OwningComponentUpdatePtr(AuthIntent.CreateAuthorityIntentUpdate().schema_type),
+							   AuthorityIntentV2::ComponentId);
+		Connection.SendComponentUpdate(EntityId, MoveTemp(Update), {});
+		MigratingEntities.Add(EntityId);
+	}
+
+	TSet<Worker_EntityId_Key> EntitiesToUpdate = EntitiesClientChanged;
+	EntitiesToUpdate.Append(EntitiesACKMigration);
+
+	for (auto EntityToMigrate : EntitiesACKMigration)
+	{
+		AuthorityIntentV2& AuthIntent = AuthorityIntentView.FindChecked(EntityToMigrate);
+		AuthorityDelegation& AuthDelegation = AuthorityDelegationView.FindChecked(EntityToMigrate);
+
+		AuthDelegation.Delegations.Add(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID, AuthIntent.PartitionId);
+		MigratingEntities.Remove(EntityToMigrate);
+	}
+	EntitiesACKMigration.Empty();
+	for (auto EntityToUpdateOwner : EntitiesClientChanged)
```

Comment:
Shame that the strategy worker has to manage client actor ownership, I would think its outside its obvious responsibilities.

Reply:
There's only one authority component :(